src/Client/StoreOnChainWallet.php

```diff
@@ -0,0 +1,252 @@
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\StoreOnChainWalletFeeRate(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getStoreOnChainWalletAddress(
+        string $storeId,
+        string $cryptoCode,
+        ?array $metaData
+    ): \BTCPayServer\Result\StoreOnChainWalletAddress {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/address';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        // Prepare metadata.
+        $metaDataMerged = [];
+
+        // Set metaData if any.
+        if ($metaData) {
+            $metaDataMerged = $metaData;
+        }
+
+        $body = json_encode(
+            [
+                'metadata' => !empty($metaDataMerged) ? $metaDataMerged : null,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return new \BTCPayServer\Result\StoreOnChainWalletAddress(
+                json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR)
+            );
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function unReserveLastStoreOnChainWalletAddress(
+        string $storeId,
+        string $cryptoCode
+    ): bool {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/address';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'DELETE';
+
+        $response = $this->getHttpClient()->request($method, $url, $headers);
+
+        if ($response->getStatus() === 200) {
+            return true;
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    public function getStoreOnChainWalletTransactions(
+        string $storeId,
+        string $cryptoCode,
+        ?array $metaData
+    ): \BTCPayServer\Result\StoreOnChainWalletTransactionList {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/payment-methods' . '/OnChain' . '/' .
+                    urlencode($cryptoCode) . '/wallet' . '/transactions';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        // Prepare metadata.
+        $metaDataMerged = [];
```

Comment:
seems same like above, not sure if query params work that way, did you find that somewhere else? I found something in the Invoice client but it is for POST request, in this case you need to append them to the url like ?param1=x&param2=y

Reply:
Alright I will make this change. Yes I copied it from the invoice client.